lib/utils/get-module-path-for.js

```diff
@@ -18,12 +18,32 @@ for (let packagePath of packagePaths) {
   let packageDir = path.dirname(path.resolve(cwd, packagePath));
 
   if (pkg.keywords && pkg.keywords.includes('ember-addon')) {
-    ADDON_PATHS[packageDir] = pkg.name;
+    // build addon instance
+    ADDON_PATHS[packageDir] = getAddonPackageName(packagePath);
   } else if (isEmberCliProject(pkg)) {
     APP_PATHS[packageDir] = pkg.name;
   }
 }
 
+/**
+ * takes a package path and returns the runtime name of the addon.
+ *
+ * @param {String} packagePath the path on disk (from current working directory)
+ * @returns {String} The runtime name of the addon
+ */
+function getAddonPackageName(packagePath) {
+  const indexPath = path.resolve(path.dirname(packagePath), 'index.js');
+
+  if (fs.existsSync(indexPath)) {
+    const { name: packageName, moduleName } = require(indexPath);
+    // this is bad, fix later
+    return moduleName && typeof moduleName === 'function' ? moduleName() : packageName;
```

Comment:
We should probably do this in a `try`/`catch` block. In most cases that I've seen, the `moduleName` function is simple (e.g. it returns a string directly, without logic) but in some cases it is reasonable to access other context off of the addon instance (e.g. `this.name`). If an addon's `moduleName` implementation does this it will error and break telemetry gathering.